core/stat/stat_slot.go

```diff
@@ -26,8 +27,38 @@ const (
 
 var (
 	DefaultSlot = &Slot{}
+
+	passCounter = metric_exporter.NewCounter(
+		"pass",
+		"Total pass count",
+		[]string{"resource"})
+	blockCounter = metric_exporter.NewCounter(
+		"block",
+		"Total block count",
+		[]string{"resource", "block_type"})
+	completeCounter = metric_exporter.NewCounter(
+		"complete",
+		"Total complete count",
+		[]string{"resource"})
+	errorCounter = metric_exporter.NewCounter(
+		"error",
+		"Total error count",
+		[]string{"resource"})
+	rtHistogram = metric_exporter.NewHistogram(
+		"rt",
+		"Rt histogram",
+		[]float64{1.0, 5.0, 10.0, 50.0, 100.0, 500.0, 1000, 5000},
+		[]string{"resource"})
 )
 
+func init() {
+	metric_exporter.MustRegister(passCounter)
```

Comment:
Here the process may panic if an error occurred during metric registering. Actually the metric exporter is not mandatory, so maybe we could provide a better mechanism to enable the exporter and handling errors?

Reply:
We can ignore the register result for now.